### ISSUE TITLE
"Unable to locate finder for 'pip._vendor.distlib'" in Python 3.6

### ISSUE THREAD

**Issue by jamesw966** [linkedin/pygradle#129]
I'm on Windows, using Python 3.6 and PyGradle 0.4.7. When I run `gradlew build` I get the following error:

```
Processing c:\users\administrator\.gradle\caches\modules-2\files-2.1\pypi\pbr\1.8.0\cc7bfd10c2031a87881bf1c92b2b03f565c76f08\pbr-1.8.0.tar.gz
Building wheels for collected packages: pbr
  Running setup.py bdist_wheel for pbr
  Stored in directory: C:\Users\Administrator\AppData\Local\pip\Cache\wheels\13\f0\a9\2ef1f08c899f898b99ec457a8d36d78e3bac314758f87c3bcc
Successfully built pbr
Installing collected packages: pbr
Exception:
Traceback (most recent call last):
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\basecommand.py", line 211, in main
    status = self.run(options, args)
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\commands\install.py", line 311, in run
    root=options.root_path,
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\req\req_set.py", line 646, in install
    **kwargs
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\req\req_install.py", line 803, in install
    self.move_wheel_files(self.source_dir, root=root)
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\req\req_install.py", line 998, in move_wheel_files
    isolated=self.isolated,
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\wheel.py", line 479, in move_wheel_files
    maker.make_multiple(['%s = %s' % kv for kv in console.items()])
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\_vendor\distlib\scripts.py", line 364, in make_multiple
    filenames.extend(self.make(specification, options))
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\_vendor\distlib\scripts.py", line 353, in make
    self._make_script(entry, filenames, options=options)
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\_vendor\distlib\scripts.py", line 257, in _make_script
    self._write_script(scriptnames, shebang, script, filenames, ext)
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\_vendor\distlib\scripts.py", line 193, in _write_script
    launcher = self._get_launcher('t')
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\_vendor\distlib\scripts.py", line 332, in _get_launcher
    result = finder(distlib_package).find(name).bytes
  File "C:\Users\Administrator\gitstuff\xim\build\venv\lib\site-packages\pip\_vendor\distlib\resources.py", line 324, in finder
    raise DistlibException('Unable to locate finder for %r' % package)
pip._vendor.distlib.DistlibException: Unable to locate finder for 'pip._vendor.distlib'
:installSetupRequirements FAILED
```

According to https://github.com/pypa/pip/issues/3964 this was fixed in pip 9.0.1, so it may be worth updating the version from 7.1.2.

**Comment by zvezdan** [comment]
To use Python 3.6 we need to update virtualenv, pip, pex, ...
That causes some issues with older packages though.
We're looking at this and testing the best combination of versions internally.
Thank you for reporting.

**Comment by slim-shah** [comment]
Hi there, i m having same problem . I want to upgrade my pip but i am unable to do it moreover i am not able to install other library due to same error

**Comment by zvezdan** [comment]
We've done a lot of work to adapt to the changes of setup, build, test requirements, such as virtualenv, setuptools, pip, and pex. The commits here prepared the ground for the upgrade I'm testing: #128, #133, #136, #138, #139, and #141.

I'll have a PR by tomorrow.

**Comment by zvezdan** [comment]
Submitted #145 for review and merge.

**Comment by zvezdan** [comment]
Resolved in the release 0.6.1.
Please upgrade dependency on pygradle to that version.
